### PR TITLE
ZeroBuf progress object and logic

### PR DESCRIPTION
**Pull request by eile**

### CODE REVIEW

**Comment by eile** [comment]
@rdumusc FYI

File: zerobuf/data/Progress.cpp [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+namespace data
+{
+Progress::Progress( const unsigned long expected )
+    : _expected( expected )
+    , _count( 0 )
+{
+    _update();
+}
+
+Progress::Progress( const std::string& operation, unsigned long expected )
```
Comment (tribal-tec) [inline zerobuf/data/Progress.cpp]: const expected

File: zerobuf/data/Progress.h [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+namespace zerobuf
+{
+namespace data
+{
+/** Drop-in progress meter for boost::progress_display. */
+class Progress : public detail::Progress
+{
+public:
+    explicit Progress( unsigned long expected );
```
Comment (tribal-tec) [inline zerobuf/data/Progress.h]: why no size_t for all the unsigned longs?

File: zerobuf/data/Progress.cpp [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+void Progress::restart( const unsigned long expected )
+{
+    _expected = expected;
+    _count = 0;
+    _update();
+}
+
+unsigned long Progress::operator+=( unsigned long inc )
```
Comment (bilgili) [inline zerobuf/data/Progress.cpp]: there can be countdown ?

File: zerobuf/data/Progress.h (hunk 2 above)
Comment (eile) [inline zerobuf/data/Progress.h]: C&P of the boost::progress_display API

File: zerobuf/data/Progress.cpp (hunk 3 above)
Comment (eile) [inline zerobuf/data/Progress.cpp]: One is making negative progress? C&P of the boost::progress_display API, which does not have this.

File: zerobuf/data/Progress.cpp (hunk 1 above)
Comment (eile) [inline zerobuf/data/Progress.cpp]: done 2x

**Pull request merged**